include/ck/tensor_operation/gpu/device/impl/device_index_pool_bwd_impl.hpp

```diff
@@ -0,0 +1,311 @@
+                        put_kernel,
+                        dim3(arg.gridSize_),
+                        dim3(arg.blockSize_),
+                        0,
+                        arg.dout_grid_desc_,
+                        arg.p_dout_,
+                        arg.p_indices_,
+                        static_cast<DInDataType_AutomicAddPreCast*>(arg.p_workspace_),
+                        PassThrough{});
+
+                    elapsed_time += launch_and_time_kernel(
+                        stream_config,
+                        cast_kernel,
+                        dim3(arg.gridSize_),
+                        dim3(arg.blockSize_),
+                        0,
+                        ck::make_tuple(arg.din_grid_desc_),
+                        ck::make_tuple(arg.din_grid_desc_),
+                        static_cast<DInDataType_AutomicAddPreCast*>(arg.p_workspace_),
+                        arg.p_din_,
+                        UnaryConvert{});
+
+                    return elapsed_time;
+                }
+                else
+                {
+                    const auto put_kernel = kernel_put_element_1d<GridwisePutElementSet,
+                                                                  InOutGrid1dDesc,
+                                                                  DOutDataType,
+                                                                  IndexDataType,
+                                                                  DInDataType,
+                                                                  PassThrough>;
+
+                    return launch_and_time_kernel(stream_config,
+                                                  put_kernel,
+                                                  dim3(arg.gridSize_),
+                                                  dim3(arg.blockSize_),
+                                                  0,
+                                                  arg.dout_grid_desc_,
+                                                  arg.p_dout_,
+                                                  arg.p_indices_,
+                                                  arg.p_din_,
+                                                  PassThrough{});
+                }
+            }
+        }
+
+        float Run(const BaseArgument* p_arg,
+                  const StreamConfig& stream_config = StreamConfig{}) override
+        {
+            return Run(*dynamic_cast<const Argument*>(p_arg), stream_config);
+        }
+    };
+
+    size_t GetWorkSpaceSize(const BaseArgument* pArg) const override
+    {
+        const Argument* pArg_ = dynamic_cast<const Argument*>(pArg);
+
+        bool needCast = pArg_->windowOverlap_ &&
+                        !(is_same_v<DInDataType, float> || is_same_v<DInDataType, double>);
+
+        if(!needCast)
+            return 0;
+        else
+        {
+            index_t din_length = pArg_->din_grid_desc_.GetTransforms()[I0].GetUpperLengths()[I0];
```

Comment:
Don't use `GetTransforms`, which is hard to understand.  Just store `din_length` in the Argument object if you need to use it in other interfaces

Reply:
47ac3767be7489134d96771ec9b214a50e7fbfc1